src/components/CreateComment.jsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from "react";
+import { postComment } from "../api";
+import { UsernameContext } from "../contexts/UsernameProvider";
+import { useContext } from "react";
+
+const CreateComment = ({article_id,setComments}) => 
+    {
+    const{username,setUsername}=useContext(UsernameContext)
+    const [searchTerm,setSearchTerm] = useState("")
+        console.log(article_id)
+    const hanldeInput=(e)=>{
+        setSearchTerm(e.target.value)
+    }
+  
+    const handeSubmit = (e)=>{
+        e.preventDefault()
+        setSearchTerm("")
+        const temporaryComment= {article_id: Date.now(),
+            author:  username,
+            body:  searchTerm,
+            comment_id : Date.now(),
+            created_at : "2024-12-17T23:52:06.394Z",
+            votes :  0}
+        setComments(currComments=>{
+            return [...currComments,temporaryComment]
+        })
+        postComment(article_id,username,searchTerm)
+        .catch(err=>{
+            console.log(err)
+            setComments(currComments=>{
+                return currComments.slice(0,-1)
+        })
+        })
```

Comment:
Doing this optimistically is not wrong, but you can also do this pessimisticlly by only adding the comment if the post request is successful, but in either case the user should be informed in some way that their comment did not get added.